buildSrc/src/main/java/org/elasticsearch/gradle/plugin/PluginPropertiesTask.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.plugin;
+
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Creates a plugin descriptor.
+ */
+public class PluginPropertiesTask extends DefaultTask {
```

Comment:
We don't want this class back now, it will go unused now as the functionality was rolled into the plugin in the mean time. 
Converting `PluginPropertiesExtension` to java is what we need to keep from this PR.

Reply:
@atorok trying to run the test or even `./gradlew build` or `check` or `etc` results in 
```
Caused by: groovy.lang.GroovyRuntimeException: Cannot get the value of write-only property 'hasNativeController' for object of type org.elasticsearch.gradle.plugin.PluginPropertiesExtension.
```  
I have changed the `hasNativeController` to `getHasNativeController` and the exception disappears. @rjernst had suggested this change (to keep it `hasNativeController`) however, since rebase this issue come up. 

For now, I have changed it to `getHasNativeController`. I have tried running the tests as follow 

```
./gradlew :build-tools:test "-Dtests.class=org.elasticsearch.gradle.plugin.PluginProprtiesExtensionTests"
``` 
but, since rebase the output is different. There is no signs of tests being run 😕

Reply:
we switched to running tests with the Gradle test runner, so the syntax to filter tests to be ran is now `./gradlew :build-tools:test --tests=*PluginProprtiesExtensionTests`

Reply:
Thanks. The docs doesn't reflect the change 🤔